extensions/interactions/base.py

```diff
@@ -89,6 +89,11 @@ class BaseInteraction(object):
     # containing this interaction. These should correspond to names of files in
     # feconf.DEPENDENCIES_TEMPLATES_DIR. Overridden in subclasses.
     _dependency_ids = []
+    # Additional JS library dependencies that should be loaded in pages
```

Comment:
Ok, so on backend part I don't think we will need any changes. What I am worried about if prediction service registry. Let's say that we are injecting all prediction service API in prediction service regisrty. Now if code interaction is not present then `CodeReplPredictionService` will also be missing. So my question is would factory still work normally even if it does not find that service in frontend or will it raise exceptions? If it raises exception then we can't inject deps hardcoded way and must think of injecting them using `$injector`, dynamically. So for backend we don't need any changes but I am not sure about frontend.